bin/propolis-server/src/lib/initializer.rs

```diff
@@ -98,46 +76,65 @@ pub fn build_instance(
         "dev64",
     )?;
 
-    Ok(Instance::create(builder.finalize()?))
+    Ok(builder.finalize()?)
 }
 
-pub struct RegisteredChipset(Arc<I440Fx>, EntityID);
+pub struct RegisteredChipset {
+    chipset: Arc<dyn Chipset>,
+    isa: Arc<i440fx::Piix3Lpc>,
+}
 impl RegisteredChipset {
-    pub fn device(&self) -> &Arc<I440Fx> {
-        &self.0
+    pub fn pci_attach(&self, bdf: pci::Bdf, dev: Arc<dyn pci::Endpoint>) {
+        self.chipset.pci_attach(bdf, dev, self.isa.route_lintr(bdf));
+    }
+    pub fn irq_pin(&self, irq: u8) -> Option<Box<dyn intr_pins::IntrPin>> {
+        self.isa.irq_pin(irq)
+    }
+    fn reset_pin(&self) -> Arc<dyn intr_pins::IntrPin> {
+        self.chipset.reset_pin()
     }
 }
 
 struct StorageBackendInstance {
     be: Arc<dyn block::Backend>,
-    child: inventory::ChildRegister,
     crucible: Option<(uuid::Uuid, Arc<block::CrucibleBackend>)>,
 }
 
 pub struct MachineInitializer<'a> {
-    log: slog::Logger,
-    machine: &'a Machine,
-    inv: &'a Inventory,
-    spec: &'a InstanceSpecV0,
-    producer_registry: Option<ProducerRegistry>,
+    pub(crate) log: slog::Logger,
+    pub(crate) machine: &'a Machine,
```

Comment:
Not necessary for this PR, just a forward-looking question: how challenging do you think it'd be to have a fake Machine? I'm thinking about what it'd take to write unit tests of the form "I added this initializer function and expect that when I call it with spec X the appropriate things will show up in maps A, B, and C."

Reply:
I think that as the core abstractions become more polished, such a thing will become easier.  `Machine` is obviously doing much less (as of this change) than `Instance` -- mostly just holding references to top-level items.  It seems likely that with a bit more clean-up (particularly around memory maps, and mediating access to the VmmHdl) that `Machine` could be discarded as well.